contracts/ExampleNFT-v2.cdc

```diff
@@ -0,0 +1,365 @@
+                    )
+                case Type<MetadataViews.Serial>():
+                    return MetadataViews.Serial(
+                        self.id
+                    )
+                case Type<MetadataViews.Royalties>():
+                    return MetadataViews.Royalties(
+                        self.royalties
+                    )
+                case Type<MetadataViews.ExternalURL>():
+                    return MetadataViews.ExternalURL("https://example-nft.onflow.org/".concat(self.id.toString()))
+                case Type<MetadataViews.NFTCollectionData>():
+                    return ExampleNFT.getCollectionData(nftType: Type<@ExampleNFT.NFT>())
+                case Type<MetadataViews.NFTCollectionDisplay>():
+                    return ExampleNFT.getCollectionDisplay(nftType: Type<@ExampleNFT.NFT>())
+                case Type<MetadataViews.Traits>():
+                    // exclude mintedTime and foo to show other uses of Traits
+                    let excludedTraits = ["mintedTime", "foo"]
+                    let traitsView = MetadataViews.dictToTraits(dict: self.metadata, excludedNames: excludedTraits)
+
+                    // mintedTime is a unix timestamp, we should mark it with a displayType so platforms know how to show it.
+                    let mintedTimeTrait = MetadataViews.Trait(name: "mintedTime", value: self.metadata["mintedTime"]!, displayType: "Date", rarity: nil)
+                    traitsView.addTrait(mintedTimeTrait)
+
+                    // foo is a trait with its own rarity
+                    let fooTraitRarity = MetadataViews.Rarity(score: 10.0, max: 100.0, description: "Common")
+                    let fooTrait = MetadataViews.Trait(name: "foo", value: self.metadata["foo"], displayType: nil, rarity: fooTraitRarity)
+                    traitsView.addTrait(fooTrait)
+                    
+                    return traitsView
+
+            }
+            return nil
+        }
+    }
+
+    pub resource Collection: NonFungibleToken.Collection, NonFungibleToken.Provider, NonFungibleToken.Receiver, NonFungibleToken.CollectionPublic, MetadataViews.ResolverCollection {
+        /// dictionary of NFT conforming tokens
+        /// NFT is a resource type with an `UInt64` ID field
+        access(contract) var ownedNFTs: @{UInt64: ExampleNFT.NFT{NonFungibleToken.NFT}}
+
+        /// Paths where this collection should be stored and linked
+        pub let defaultStoragePath: StoragePath
+        pub let defaultPublicPath: PublicPath
+        pub let privateProviderPath: PrivatePath
+
+        init () {
+            self.ownedNFTs <- {}
+            self.defaultStoragePath = /storage/cadenceExampleNFTCollection
+            self.defaultPublicPath = /public/cadenceExampleNFTCollection
+            self.privateProviderPath = /private/cadenceExampleNFTCollection
+        }
+
+        /// Returns the NFT types that this collection can store
+        pub fun getAcceptedTypes(): {Type: Bool} {
```

Comment:
```suggestion
        pub fun getAcceptedTypes(): [Type] {
```

Reply:
Okey not really, `Receiver` and `CollectionPublic` both have a `getAcceptedTypes()` method, are both needed? Probably just returning `[Type]` will do the trick right?

Reply:
We'd discussed this briefly somewhere before

If we make this a list that means we are basically forcing people to loop over an array to see if something is supported. Using a dictionary makes that a non-issue, I would argue that the small bloat in data is worth it, especially since you can just get the keys of the response if an array is really what you want. This convention also lets you use the dictionary to mark what **isn't** supported which is some nice flexibility.

Reply:
Thanks @austinkline !!